### PR TITLE
feat(pack): adds angular pack

### PR DESCRIPTION
**Pull request by axelcalixte**
Hello,

I noticed there wasn't any pack for Angular so I made one for myself.
Maybe it is useful to someone else.

### CODE REVIEW

**Comment by luxus** [comment]
does it make sense that it depends on the ts pack? @owittek @Uzaaft

**Comment by Uzaaft** [comment]
I don't think so. I remember mehalter voting against it whenever @owittek  did it in the deno pack.

**Comment by owittek** [comment]
I actually do the same with the all-in-one pack and since Angular is made for Node I think it's fine to depend on the default ts-pack.

**Review by mehalter — CHANGES_REQUESTED** [review]
LGTM! We just need to mention the angular treesitter plugin that gets installed. Is that a dependency for treesitter?

**Comment by owittek** [comment]
We might want to switch the branch for the plugin as described here: https://github.com/nvim-treesitter/nvim-treesitter-angular/pull/4#issuecomment-1372982890

**Comment by axelcalixte** [comment]
> LGTM! We just need to mention the angular treesitter plugin that gets installed. Is that a dependency for treesitter?

@mehalter The angular treesitter parser isn't available via `:TSInstall angular`. That's why the plugin is needed.

> We might want to switch the branch for the plugin as described here: [nvim-treesitter/nvim-treesitter-angular#4 (comment)](https://github.com/nvim-treesitter/nvim-treesitter-angular/pull/4#issuecomment-1372982890)

@owittek  I wasn't aware of this branch. Thank you !

**Review by owittek — APPROVED** [review]
LGTM! @Uzaaft @luxus We'll have to refactor when we add a js pack as discussed on discord

**Comment by mehalter** [comment]
@owittek do you mean `js` or `html+css` web pack? I was saying on discord a minimal html+css pack without any JavaScript

**Comment by mehalter** [comment]
I can do the refactor on Monday

**Comment by owittek** [comment]
> @owittek do you mean `js` or `html+css` web pack? I was saying on discord a minimal html+css pack without any JavaScript 

I already replied on discord but I meant the `js pack` with a working DAP which should be added at the same time as the `html+css` web pack. I just wanted to mention it since it would have to be refactored too.

File: lua/astrocommunity/pack/angular/README.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Angular Language Pack
+
+This plugin pack does the following:
+
+- Adds [tsserver pack](https://github.com/AstroNvim/astrocommunity/tree/main/lua/astrocommunity/pack/typescript)
+- Adds `[nvim-treesitter-angular](https://github.com/elgiano/nvim-treesitter-angular)` for `angular` Treesitter parser
+- Adds `html` and `css` Treesitter parsers
```
Comment (owittek) [inline lua/astrocommunity/pack/angular/README.md:7]:
```suggestion
- Adds [HTML & CSS support](../html-css)
```

File: lua/astrocommunity/pack/angular/angular.lua [hunk 2]
```diff
@@ -0,0 +1,22 @@
+local utils = require "astronvim.utils"
+
+return {
+  {
+    import = "astrocommunity.pack.typescript",
+  },
```
Comment (owittek) [inline lua/astrocommunity/pack/angular/angular.lua:6]:
```suggestion
  { import = "astrocommunity.pack.html-css" },
  { import = "astrocommunity.pack.typescript" },
```

File: lua/astrocommunity/pack/angular/angular.lua [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  { "elgiano/nvim-treesitter-angular", branch = "topic/jsx-fix" },
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if opts.ensure_installed ~= "all" then
+        opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "html", "css" })
+      end
+    end,
+  },
```
Comment (owittek) [inline lua/astrocommunity/pack/angular/angular.lua:15]:
```suggestion
```

File: lua/astrocommunity/pack/angular/angular.lua [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  {
+    "williamboman/mason-lspconfig.nvim",
+    opts = function(_, opts)
+      opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, { "angularls", "html", "cssls" })
+    end,
```
Comment (owittek) [inline lua/astrocommunity/pack/angular/angular.lua:20]:
```suggestion
    opts = function(_, opts) opts.ensure_installed = utils.list_insert_unique(opts.ensure_installed, "angularls") end,
```

**Review by owittek — CHANGES_REQUESTED** [review]
Refactor to use the html-css base pack

**Review by mehalter — APPROVED** [review]
Did a refactor to move to using the `html-css` pack and also moved the `elgiano/nvim-treesitter-angular` to be a dependency of `nvim-treesitter`. Have tested locally and looks good to go! Thanks @axelcalixte !

**Pull request merged**